File: scripts/reference-solutions/employee-free-time.cpp
#include <algorithm>
#include <iostream>
#include <utility>
#include <vector>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  int employees;
  if (!(cin >> employees)) {
    cout << "EMPTY\n";
    return 0;
  }

  vector<pair<long long, long long>> busy;
  for (int employee = 0; employee < employees; ++employee) {
    int count;
    cin >> count;
    for (int i = 0; i < count; ++i) {
      long long start, end;
      cin >> start >> end;
      busy.push_back({start, end});
    }
  }

  if (busy.empty()) {
    cout << "EMPTY\n";
    return 0;
  }

  sort(busy.begin(), busy.end());
  vector<pair<long long, long long>> freeIntervals;
  long long currentEnd = busy[0].second;

  for (int i = 1; i < static_cast<int>(busy.size()); ++i) {
    const auto& [start, end] = busy[i];
    if (start > currentEnd) {
      freeIntervals.push_back({currentEnd, start});
      currentEnd = end;
    } else {
      currentEnd = max(currentEnd, end);
    }
  }

  if (freeIntervals.empty()) {
    cout << "EMPTY\n";
  } else {
    for (const auto& [start, end] : freeIntervals) {
      cout << start << ' ' << end << '\n';
    }
  }

  return 0;
}

File: scripts/reference-solutions/employee-free-time.py
import sys

tokens = sys.stdin.read().strip().split()
if not tokens:
    print("EMPTY")
    sys.exit(0)

index = 0
employees = int(tokens[index])
index += 1
busy = []

for _ in range(employees):
    count = int(tokens[index])
    index += 1
    for _ in range(count):
        start = int(tokens[index])
        end = int(tokens[index + 1])
        index += 2
        busy.append((start, end))

if not busy:
    print("EMPTY")
    sys.exit(0)

busy.sort()
free = []
current_start, current_end = busy[0]

for start, end in busy[1:]:
    if start > current_end:
        free.append((current_end, start))
        current_start, current_end = start, end
    else:
        current_end = max(current_end, end)

if free:
    print("\n".join(f"{start} {end}" for start, end in free))
else:
    print("EMPTY")
